server/midddleware/request_metadata.go

```
@@ -0,0 +1,101 @@
+	"github.com/tigrisdata/tigris/server/transaction"
+)
+
+type RequestMetadataCtxKey struct {
+}
+
+type AccessToken struct {
+	Namespace string
+	Sub       string
+}
+
+type RequestMetadata struct {
+	accessToken AccessToken
+	namespace   metadata.Namespace
+}
+
+// NamespaceExtractor - extract the namespace from context
+type NamespaceExtractor interface {
+	Extract(ctx context.Context) (metadata.Namespace, error)
+}
+
+type AccessTokenNamespaceExtractor struct {
+	TenantManager      *metadata.TenantManager
+	Transactionmanager *transaction.Manager
+}
+
+func GetRequestMetadata(ctx context.Context) (*RequestMetadata, error) {
+	// read token
+	value := ctx.Value(RequestMetadataCtxKey{})
+	if value != nil {
+		if requestMetadata, ok := value.(*RequestMetadata); ok {
+			return requestMetadata, nil
+		}
+	}
+	return nil, api.Errorf(api.Code_NOT_FOUND, "request metadata not found 1")
+}
+
+func setAccessToken(ctx context.Context, token AccessToken) context.Context {
+	requestMetadata, err := GetRequestMetadata(ctx)
+	var result = ctx
+	if err != nil && requestMetadata == nil {
+		requestMetadata = &RequestMetadata{}
+		result = context.WithValue(ctx, RequestMetadataCtxKey{}, requestMetadata)
+	}
+	requestMetadata.accessToken = token
+	return result
+}
+
+func setNamespace(ctx context.Context, namespace metadata.Namespace) context.Context {
+	requestMetadata, err := GetRequestMetadata(ctx)
+	var result = ctx
+	if err != nil && requestMetadata == nil {
+		requestMetadata = &RequestMetadata{}
+		result = context.WithValue(ctx, RequestMetadataCtxKey{}, requestMetadata)
+	}
+	requestMetadata.namespace = namespace
+	return result
+}
+func GetAccessToken(ctx context.Context) (*AccessToken, error) {
+	// read token
+	value := ctx.Value(RequestMetadataCtxKey{})
+	if value != nil {
+		if requestMetadata, ok := value.(*RequestMetadata); ok {
+			return &requestMetadata.accessToken, nil
+		}
+	}
+	return nil, api.Errorf(api.Code_NOT_FOUND, "request metadata not found 2")
```

If we need to return an error from here then it should be that `access token not found`.

---

fixed. (it was left over from debugging)